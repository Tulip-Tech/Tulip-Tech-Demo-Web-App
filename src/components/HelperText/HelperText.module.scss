@use 'src/styles/variables';
@use 'src/styles/theme';

.helperText {
  margin-top: 4px;
  font-size: 12px;
  text-align: left;

  &.error {
    color: theme.$text-field-error-color;
  }
}
