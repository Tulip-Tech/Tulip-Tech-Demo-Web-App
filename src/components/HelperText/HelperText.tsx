import React from 'react';
import classNames from 'classnames';

import styles from './HelperText.module.scss';

type Props = {
  children: React.ReactNode;
  className?: string;
  error?: boolean;
};

const HelperText: React.FC<Props> = ({ children, className, error }: Props) => {
  return children ? <div className={classNames(styles.helperText, { [styles.error]: error }, className)}>{children}</div> : null;
};

export default HelperText;
