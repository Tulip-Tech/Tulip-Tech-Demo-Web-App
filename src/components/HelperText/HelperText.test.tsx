import React from 'react';
import { render } from '@testing-library/react';

import HelperText from './HelperText';

describe('<HelperText>', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<HelperText error={true}>Hello world!</HelperText>);

    expect(container).toMatchSnapshot();
  });
});
