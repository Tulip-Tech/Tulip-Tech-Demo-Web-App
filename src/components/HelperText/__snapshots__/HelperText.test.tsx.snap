// Vitest Snapshot v1

exports[`<HelperText> > renders and matches snapshot 1`] = `
<div>
  <div
    class="_helperText_da58e0 _error_da58e0"
  >
    Hello world!
  </div>
</div>
`;
